sdk/identity/src/token_credentials/azure_cli_credentials.rs

```diff
@@ -94,18 +95,47 @@ mod az_cli_date_format {
     }
 }
 
+/// The respnose from `az account get-access-token --output json`.
 #[derive(Debug, Clone, Deserialize)]
-#[serde(rename_all = "camelCase")]
 struct CliTokenResponse {
+    #[serde(rename = "accessToken")]
     pub access_token: Secret,
-    #[serde(with = "az_cli_date_format")]
+    #[cfg(feature = "old_azure_cli")]
```

Comment:
Instead of a feature, can we just check if it exists in the response and use it?

Reply:
Yes, if it is in the response, it uses it.

The feature `old_azure_cli` affects when `expire_on` is not in the response. Additional dependencies are needed for `assume_local`. If you remove the `old_azure_cli` feature, it removes those dependencies, but only works with AZ CLI 2.54.0 and above.

Reply:
This PR has two changes.
1. use `expires_on` if it exists
2. the ability to opt out of the additional local timezone dependencies by not including `old_azure_cli`